osmoutils/binary_search.go

```diff
@@ -56,6 +58,44 @@ func (e ErrTolerance) Compare(expected sdk.Int, actual sdk.Int) int {
 	return 0
 }
 
+// CompareBigDec validates if actual is within errTolerance of expected.
+// returns 0 if it is
+// returns 1 if not, and expected > actual.
+// returns -1 if not, and expected < actual
+func (e ErrTolerance) CompareBigDec(expected osmomath.BigDec, actual osmomath.BigDec) int {
+	diff := expected.Sub(actual).Abs()
+
+	comparisonSign := 0
+	if expected.GT(actual) {
+		comparisonSign = 1
+	} else {
+		comparisonSign = -1
+	}
+
+	// Check additive tolerance equations
+	if !e.AdditiveTolerance.IsNil() {
+		// if no error accepted, do a direct compare.
+		if e.AdditiveTolerance.IsZero() {
+			if expected.Equal(actual) {
+				return 0
+			}
+		}
+
+		if diff.GT(osmomath.NewBigDec(e.AdditiveTolerance.Int64())) {
```

Comment:
I don't think this quite makes sense, to increase the precision, we first truncate to int64 and then increase the size.

Should probably e.AdditiveTolerance.ToDec() and then make a big dec from that.

Reply:
Ah so we didn't have the `BigDecFromSDKDec` implemented in `BigDec` when I made this PR so what you suggested wasn't possible then, but I actually agree with this change! Updating now